apollo-router/tests/samples/basic/requires/plan.json

```diff
@@ -0,0 +1,224 @@
+                                            }
+                                        ]
+                                    }
+                                }
+                            }
+                        },
+                        {
+                            "request": {
+                                "body": {
+                                    "query": "query TestD__products__2($representations:[_Any!]!){_entities(representations:$representations){...on D{j}}}",
+                                    "operationName": "TestD__products__2",
+                                    "variables": {
+                                        "representations": [
+                                            {
+                                                "__typename": "D",
+                                                "i": [
+                                                    {
+                                                        "x": "xy",
+                                                        "y": "y"
+                                                    },
+                                                    {
+                                                        "x": "xz",
+                                                        "z": "z"
+                                                    }
+                                                ],
+                                                "id": "1"
+                                            }
+                                        ]
+                                    }
+                                }
+                            },
+                            "response": {
+                                "body": {
+                                    "data": {
+                                        "_entities": [
+                                            {
+                                                "j": "j"
+                                            }
+                                        ]
+                                    }
+                                }
+                            }
+                        }
+                    ]
+                }
+            }
+        },
+        {
+            "type": "Request",
+            "request": {
+                "query": "query TestA { a { c { c } } }"
+            },
+            "expected_response": {
+                "data": {
+                    "a": {
+                        "c": {
+                            "c": "c"
+                        }
+                    }
+                }
+            }
+        },
+        {
+            "type": "Request",
+            "request": {
+                "query": "query TestD { d { j } }"
+            },
+            "expected_response": {
+                "data": {
+                    "d": {
+                        "j": "j"
+                    }
+                }
+            }
+        },
```

Comment:
This fails right now because with `D` defined as follows:

```graphql
type D
  @join__type(graph: ACCOUNTS, key: "id")
  @join__type(graph: REVIEWS, key: "id")
  @join__type(graph: PRODUCTS, key: "id") {
    id: ID!
    i: [I] @join__field(graph: REVIEWS) @join__field(graph: PRODUCTS, external: true)
    j: String @join__field(graph: PRODUCTS, requires: "i { x ... on XY { y } ... on XZ { z } }")
}
```

we would expect a query to `REVIEWS` asking for the x, y and z fields, but it only asks the `x` field

Reply:
here are the queries we see happening on the subgraph side:
```
  method: POST
        path: http://localhost/
        {"query":"query TestD__accounts__0{d{__typename id}}","operationName":"TestD__accounts__0"}

        method: POST
        path: http://localhost/
        {"query":"query TestD__reviews__1($representations:[_Any!]!){_entities(representations:$representations){...on D{i{__typename x}}}}","operationName":"TestD__reviews__1","variables":{"representations":[{"__typename":"D","id":"1"}]}}

        method: POST
        path: http://localhost/
        {"query":"query TestD__products__2($representations:[_Any!]!){_entities(representations:$representations){...on D{j}}}","operationName":"TestD__products__2","variables":{"representations":[{"__typename":"D","i":[{"x":"xy"},{"x":"xz"}],"id":"1"}]}}
```

I think the second query should actually be:

```graphql
query TestD__reviews__1($representations:[_Any!]!){
  _entities(representations:$representations)
  {
    ...on D {
      i{
        __typename
        x
        ... on XY {
          y
        }
        ... on XZ {
          z
        }
      }
    }
  }
}
```

Here is the generated query plan:
```json
{

  "object": {
    "kind": "QueryPlan",
    "node": {
      "kind": "Sequence",
      "nodes": [
        {
          "kind": "Fetch",
          "serviceName": "accounts",
          "variableUsages": [],
          "operation": "query TestD__accounts__0{d{__typename id}}",
          "operationName": "TestD__accounts__0",
          "operationKind": "query",
          "id": null,
          "inputRewrites": null,
          "outputRewrites": null,
          "contextRewrites": null,
          "schemaAwareHash": "15b916b342c906e6ef1b6d55fc6e52cf213f4f8d343cb535161cdb8b93ad5be7",
          "authorization": {
            "is_authenticated": false,
            "scopes": [],
            "policies": []
          }
        },
        {
          "kind": "Flatten",
          "path": [
            "d"
          ],
          "node": {
            "kind": "Fetch",
            "serviceName": "reviews",
            "requires": [
              {
                "kind": "InlineFragment",
                "typeCondition": "D",
                "selections": [
                  {
                    "kind": "Field",
                    "name": "__typename"
                  },
                  {
                    "kind": "Field",
                    "name": "id"
                  }
                ]
              }
            ],
            "variableUsages": [],
            "operation": "query TestD__reviews__1($representations:[_Any!]!){_entities(representations:$representations){...on D{i{__typename x}}}}",
            "operationName": "TestD__reviews__1",
            "operationKind": "query",
            "id": null,
            "inputRewrites": null,
            "outputRewrites": null,
            "contextRewrites": null,
            "schemaAwareHash": "6430f65befcf0b19f22c83cbf482768fc47aecf7697b831962175e5881c5fb5e",
            "authorization": {
              "is_authenticated": false,
              "scopes": [],
              "policies": []
            }
          }
        },
        {
          "kind": "Flatten",
          "path": [
            "d"
          ],
          "node": {
            "kind": "Fetch",
            "serviceName": "products",
            "requires": [
              {
                "kind": "InlineFragment",
                "typeCondition": "D",
                "selections": [
                  {
                    "kind": "Field",
                    "name": "__typename"
                  },
                  {
                    "kind": "Field",
                    "name": "i",
                    "selections": [
                      {
                        "kind": "Field",
                        "name": "x"
                      },
                      {
                        "kind": "InlineFragment",
                        "typeCondition": "XY",
                        "selections": [
                          {
                            "kind": "Field",
                            "name": "y"
                          }
                        ]
                      },
                      {
                        "kind": "InlineFragment",
                        "typeCondition": "XZ",
                        "selections": [
                          {
                            "kind": "Field",
                            "name": "z"
                          }
                        ]
                      }
                    ]
                  },
                  {
                    "kind": "Field",
                    "name": "id"
                  }
                ]
              }
            ],
            "variableUsages": [],
            "operation": "query TestD__products__2($representations:[_Any!]!){_entities(representations:$representations){...on D{j}}}",
            "operationName": "TestD__products__2",
            "operationKind": "query",
            "id": null,
            "inputRewrites": null,
            "outputRewrites": null,
            "contextRewrites": null,
            "schemaAwareHash": "a48c3f7980eb280b66742069d8dda4ceba70ebb9a4856c541f7900fa023fd48f",
            "authorization": {
              "is_authenticated": false,
              "scopes": [],
              "policies": []
            }
          }
        }
      ]
    }
  },
  "text": "QueryPlan {\n  Sequence {\n    Fetch(service: \"accounts\") {\n      {\n        d {\n          __typename\n          id\n        }\n      }\n    },\n    Flatten(path: \"d\") {\n      Fetch(service: \"reviews\") {\n        {\n          ... on D {\n            __typename\n            id\n          }\n        } =>\n        {\n          ... on D {\n            i {\n              __typename\n              x\n            }\n          }\n        }\n      },\n    },\n    Flatten(path: \"d\") {\n      Fetch(service: \"products\") {\n        {\n          ... on D {\n            __typename\n            i {\n              x\n              ... on XY {\n                y\n              }\n              ... on XZ {\n                z\n              }\n            }\n            id\n          }\n        } =>\n        {\n          ... on D {\n            j\n          }\n        }\n      },\n    },\n  },\n}"
}
```

Could it be an issue in the way I wrote the supergraph schema?

Reply:
ok, the subgraph reload was missing the `"update_url_overrides": true` option so the query was going to starstuff